bot/cogs/news.py

```diff
@@ -0,0 +1,255 @@
+        payload = existing_news.copy()
+
+        for maillist in constants.PythonNews.mail_lists:
+            async with self.bot.http_session.get(RECENT_THREADS_TEMPLATE.format(name=maillist)) as resp:
+                recents = BeautifulSoup(await resp.text(), features="lxml")
+
+            for thread in recents.html.body.div.find_all("a", href=True):
+                # We want only these threads that have identifiers
+                if "latest" in thread["href"]:
+                    continue
+
+                thread_information, email_information = await self.get_thread_and_first_mail(
+                    maillist, thread["href"].split("/")[-2]
+                )
+
+                try:
+                    new_date = datetime.strptime(email_information["date"], "%Y-%m-%dT%X%z")
+                except ValueError:
+                    log.warning(f"Invalid datetime from Thread email: {email_information['date']}")
+                    continue
+
+                if (
+                        await self.check_new_exist(thread_information["subject"], new_date, maillist, existing_news)
+                        or new_date.date() < date.today()
+                ):
+                    continue
+
+                content = email_information["content"]
+                link = THREAD_URL.format(id=thread["href"].split("/")[-2], list=maillist)
+                msg = await self.send_webhook(
+                    title=thread_information["subject"],
+                    description=content[:500] + f"... [continue reading]({link})" if len(content) > 500 else content,
+                    timestamp=new_date,
+                    url=link,
+                    author=f"{email_information['sender_name']} ({email_information['sender']['address']})",
+                    author_url=MAILMAN_PROFILE_URL.format(id=email_information["sender"]["mailman_id"]),
+                    webhook_profile_name=self.webhook_names[maillist],
+                    footer=f"Posted to {self.webhook_names[maillist]}"
+                )
+                payload["data"][maillist].append(msg.id)
+
+                if msg.channel.is_news():
+                    log.trace("Publishing mailing list message because it was in a news channel")
+                    await msg.publish()
+
+        await self.bot.api_client.put("bot/bot-settings/news", json=payload)
+
+    async def check_new_exist(self, title: str, timestamp: datetime, maillist: str, news: t.Dict[str, t.Any]) -> bool:
+        """Check does this new title + timestamp already exist in #python-news."""
+        for new in news["data"][maillist]:
+            message = discord.utils.get(self.bot.cached_messages, id=new)
+            if message is None:
+                message = await self.channel.fetch_message(new)
+                if message is None:
+                    log.trace(f"Could not find message for {new} on mailing list {maillist}")
+                    return False
```

Comment:
Fetching all the messages on every new post seems like it won't scale well. Why can't we use the ID returned from the mailing list API? `thread_id`: https://mail.python.org/archives/api/list/python-announce-list@python.org/thread/M2CB6LG65LMXR4FJTRVGTW2IWUZRUJA5/